File: doc-examples/example-jackson-cbor-java/src/main/java/example/BookController.java
package example;

import io.micronaut.http.annotation.Body;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Post;
import io.micronaut.serde.cbor.CborMediaTypes;

@Controller("/books")
public class BookController {

    @Post(processes = CborMediaTypes.APPLICATION_CBOR)
    public Book save(@Body Book book) {
        return book;
    }
}

File: doc-examples/example-jackson-cbor-python/src/main/python/example/BookController.py
from typing import Annotated

from micronaut.http.annotation import Body, Controller, Post
from micronaut.serde.cbor import CborMediaTypes

from example.Book import Book


@Controller("/books")
class BookController:

    @Post(processes=CborMediaTypes.APPLICATION_CBOR)
    def save(self, book: Annotated[Book, Body]) -> Book:
        return book
